File: java/archbots/nullmage/IsAlchSelected.java
package archbots.nullmage;

import com.runemate.game.api.osrs.local.hud.interfaces.Magic;
import com.runemate.game.api.script.framework.tree.BranchTask;
import com.runemate.game.api.script.framework.tree.TreeTask;

public class IsAlchSelected extends BranchTask {

    private AlchItem alchItem = new AlchItem();
    private IsCurseSelected isCurseSelected = new IsCurseSelected();

    @Override
    public TreeTask successTask() {
        return alchItem;
    }

    @Override
    public TreeTask failureTask() {
        return isCurseSelected;
    }

    @Override
    public boolean validate() {
        return Magic.HIGH_LEVEL_ALCHEMY.isSelected();
    }
}

File: kotlin/archbots/archmage/treebot/branches/IsAlchSelected.kt
package archbots.archmage.treebot.branches

import archbots.archmage.treebot.leaves.AlchItem
import com.runemate.game.api.osrs.local.hud.interfaces.Magic
import archapi.treebot.framework.BranchTask
import archapi.treebot.framework.TreeTask

class IsAlchSelected : BranchTask() {

    private val alchItem = AlchItem()
    private val isCurseSelected = IsCurseSelected()

    override fun successTask(): TreeTask {
        return alchItem
    }

    override fun failureTask(): TreeTask {
        return isCurseSelected
    }

    override fun validate(): Boolean {
        return Magic.HIGH_LEVEL_ALCHEMY.isSelected
    }
}
